drivers/sensors/analog_joystick.c

```diff
@@ -89,12 +89,36 @@ int16_t axisCoordinate(pin_t pin, uint16_t origin, uint8_t axis) {
 }
 
 int8_t axisToMouseComponent(pin_t pin, int16_t origin, uint8_t maxSpeed, uint8_t axis) {
-    int16_t coordinate = axisCoordinate(pin, origin, axis);
+    int32_t coordinate = axisCoordinate(pin, origin, axis);
     int8_t  result;
 #ifndef ANALOG_JOYSTICK_WEIGHTS
+    int8_t scale = 100;
+    coordinate *= scale;
     if (coordinate != 0) {
-        float percent = (float)coordinate / 100;
-        result        = percent * maxCursorSpeed * (abs(coordinate) / speedRegulator);
+        int32_t component = (coordinate / 100 * maxCursorSpeed * abs(coordinate)) / (speedRegulator * scale);
+
+        int32_t dead_zone;
+        if (pin == ANALOG_JOYSTICK_X_AXIS_PIN) {
+            dead_zone = ANALOG_JOYSTICK_X_AXIS_DEADZONE;
+        } else {
+            dead_zone = ANALOG_JOYSTICK_Y_AXIS_DEADZONE;
+        }
+        dead_zone *= scale / 100;
+
+        int32_t sign = (component > 0) ? 1 : -1;
+        if (abs(component) < dead_zone) {
+            component = 0;
+        } else if (abs(component) < scale) {
+            /* Can't return less than 1, or it will be floored to 0 by
+            integer division when we descale. To maintain
+            sensitivity for small ranges, instead boost it to a
+             minimum of 1  i f th ere is some sort of signal.
+             */
+            component = sign * scale;
```

Comment:
Note that return type of the function is `int8_t`. It can't return numbers smaller than 1.

Reply:
yes, that's the whole point of this PR. If it tries to return 1, it sets it to a minimum of 1